Update user guide (#191)
AY2223S1-CS2103T-W11-3/tp: docs/UserGuide.md

title: User Guide
---

Made for commission-based artists, **ArtBuddy** (AB) is the easiest way to organise your customers and commissions.
Powerful features and intuitive design, all packaged into one desktop app. With optimised support for
use via a Command Line Interface (CLI) ArtBuddy can help you manage your small business with the speed of your fingertips. But not to worry if you're unfamiliar with CLI, we still offer the benefits of a Graphical User Interface (GUI).

* Table of Contents
{:toc}

--------------------------------------------------------------------------------------------------------------------
# Using this guide

This guide walks you through all the features of ArtBuddy and can be used as a quick reference whenever you need any help.

> If you are a new user, We welcome you to start from our [introduction](#introduction) to get control of yourbusiness.

Reviewer: ```suggestion
If you are a new user, we welcome you to start from our [introduction](#introduction) to get control of your business.
```

2. Download the latest `artbuddy.jar` from [here](https://github.com/AY2223S1-CS2103T-W11-3/tp/releases).

3. Copy the file to the folder you want to use as the _home folder_ for your ArtBuddy.

4. Double-click the file to start the app. You can start typing to input your commands. If you click away, you can click on the top box to start typing again.
5. The GUI should display a list of customers. This view will be referred to as Customer View. <br> ![Ui](images/UiCustomerList.png)
6. You can use the following commands in the Customer View:

   * `addcus n/Betsy Crowe e/betsycrowe@example.com p/1234586` :  Creates a new customer named Betsy Crowe.
   * `delcus 2` :  Deletes the 2nd customer displayed on the list.
   * `opencus 2` : Selects the 2nd customer (Bernice Yu) on the list.

7. In the Commission View, the list of all commissions from the opened customer (Alex Yeoh) is displayed. <br> ![Ui](images/UiCommissionList.png)
8. You can use the following commands in the Commission View:
    * `addcom n/Fate Archer f/60 d/2022-10-15 s/true p/Archery t/night t/city` : Creates a new commission titled "Fate Archer" under Meg.

>     * `delcom 2` : Deletes the 5th commission from Alex Yeoh on the list.

Reviewer: ```suggestion
    * `delcom 2` : Deletes the 2nd commission from Alex Yeoh on the list.
```

* `INDEX` refers to the numbering shown on the list. <br>
  e.g. in `opencus INDEX`, `INDEX` refers to the numbering shown on the customers list.

* Anything with prefix `d/` such as `d/DATE` should be formatted in YYYY-MM-DD. <br>
  e.g. `d/2022-10-04` is a valid input.

* Items in square brackets are optional.<br>
  e.g `n/NAME [t/TAG]` can be used as `n/John Doe t/friend` or as `n/John Doe`.

* Items with `...` after them can be used multiple times including zero times.<br>
  e.g. `[t/TAG]...` can be used as ` ` (i.e. 0 times), `t/friend`, `t/friend t/family` etc.

* Parameters can be in any order.<br>
  e.g. if the command specifies `n/NAME p/PHONE_NUMBER`, `p/PHONE_NUMBER n/NAME` is also acceptable.

* If a parameter is expected only once in the command, but you specified it multiple times, only the last occurrence of the parameter will be taken.<br>
  e.g. if you specify `p/12341234 p/56785678`, only `p/56785678` will be taken.

* Extraneous parameters for commands that do not take in parameters (such as `help`, `list`, `exit` and `clear`) will be ignored.<br>
  e.g. if the command specifies `help 123`, it will be interpreted as `help`.

</div>

### Viewing help: `help`

> Shows you message explaning how to access the help page.

Reviewer: ```suggestion
Shows you a message explaining how to access the help page.
```

### Deleting a customer: `delcus`

Deletes the customer at `INDEX` from the ArtBuddy.

Format: `delcus INDEX`

Examples:
* `delcus 2` Deletes the 2nd customer in the ArtBuddy and all commissions made by the customer.

## Commission related commands

A customer may ask for multiple commissions with different needs.
That is why we created the commission to help you keep track of all the commissions.

A commission will have the following details:
* Title
* Fee
* Deadline
* Status
* Optional Description
* Optional Tags

### Viewing a commission: `opencom`

Opens a commission at `INDEX` and shows its relevant details and image. If no `INDEX` is specified, you will just switch to the commissions tab.

Format: `opencom [INDEX]`

Examples:
* `opencom` Switches to commissions tab.

* `opencom 2` When you run an `opencom` command with index `opencom 2`, you should be switched to the commissions tab to view commission details.

![opencom](images/opencom.png)

### Adding a commission: `addcom`

Adds a commission to the currently opened customer.

Format: `addcom n/TITLE f/FEE d/DEADLINE s/STATUS [p/DESCRIPTION] [t/TAG]...​`

> * Fee is the fee you are charging the customer for this commission.
> * Status states if the commission is done or not and can only take `y`, `yes`, `t`, `true` for complete status and `n`, `no`, `f`, `false` for incomplete status case insensitive.

Reviewer: ```suggestion
* Fee is the amount you are charging the customer for this commission.
* Status states if the commission is done or not and can only take `y`, `yes`, `t`, `true` for complete status and `n`, `no`, `f`, `false` for incomplete status. It is case insensitive.
```

Format: `find [k/KEYWORDS]... -all [t/TAGS]... -any [t/TAGS]...`
* The keywords, all, and any filters are all optional and can be omitted, but at least one should exist.

Examples:
* `find k/Kevin` Finds customers who have name Kevin.
* `find -all t/friend t/colleague` Finds customers who are tagged both `friend` and `colleague`.
* `find -any t/friend t/colleague` Finds customers who are either tagged `friend` or `colleague`.

### Sorting the customer list: `sortcus`

Sorts the displayed customer list by one of the following options:

Prefix:
- `n` (name)
- `d` (latest commission date)
- `c` (commission count)
- `r` (revenue)
- `a` (active commissions count)

Suffix: `+` (increasing) or `-` (decreasing)

Format: `sortcus PREFIX/SUFFIX`

Examples:
* `sortcus n/+` Sorts the customer list from A to Z.

### List the commissions: `listcom`
Lists all the commissions made by selected customer.

Format: `listcom`

### Viewing all commissions: `allcom`

Displays all commissions across all customers in ArtBuddy.

Format: `allcom`

After running `allcom`, you should be switched to the commissions tab to view all commissions.

> To view the commissions for a specific customer, return to the customer list ([`opencus`](#opencus)) and select the customer from the list ([`opencus INDEX`](#opencus)).

Reviewer: Can you help fix these links? They don't work. Thanks!

### Find a commission: `findcom`
Finds all the commissions in the list which satisfy keyword matching in the title and tag filters. The tag filter more specifically will include commissions which contain all tags under `-all` and at least one of the tags under `-any`.

Format: `find [k/KEYWORDS]... -all [t/TAGS]... -any [t/TAGS]...`
* The keywords, all, and any filters are all optional and can be omitted, but at least one should exist.

Examples:
* `find k/Kevin` Finds customers who have name Kevin.
* `find -all t/friend t/colleague` Finds customers who are tagged both `friend` and `colleague`.
* `find -any t/friend t/colleague` Finds customers who are either tagged `friend` or `colleague`.

## Miscellaneous
 
### Clearing all the customers: `clear`

> Clears every customer in the ArtBuddy

Reviewer: ```suggestion
Clears all your customers from ArtBuddy, including their commissions and iterations.
```

Technically, because of how we store our data, commissions and iterations would also be cleared as well :P

**Edit customer** | `editcus INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...` <br> e.g., `editcus 1 p/91234567 e/johndoe@example.com`
**Delete customer** | `delcus INDEX`<br> e.g., `delcus 14`
**Open Commission** | `opencom INDEX`<br> e.g., `opencom 14`
**Add commission** | `addcom n/TITLE f/FEE d/DEADLINE [t/TAG]...`<br> e.g., `addcom n/Rimuru f/40 d/2022-11-01 t/traditional t/chibi`
**Edit commission** | `editcom INDEX [n/TITLE] [f/FEE] [d/DEADLINE] [s/COMPLETION STATUS] [p/DESCRIPTION] [t/TAG]...` <br> e.g., `editcom 1 n/Tokyo Ghoul Kaneki f/50 d/2022-10-10 s/False p/Unfamiliar, I will need to do up a reference board first. t/digital t/neon`
**Delete Commission** | `delcom INDEX`<br> e.g., `delcom 14`
**Add Iteration** | `additer n/DESCRIPTION d/DATE f/FEEDBACK p/FILEPATH`<br> e.g., `additer n/Draft 1 f/Good d/ 2022-10-28 p//Users/John/Downloads/Bread.jpeg`
**Edit Iteration**| `editer INDEX [n/DESCRIPTION] [d/DATE] [f/FEEDBACK] [p/FILEPATH]`<br> e.g, `edititer 2 n/Sketch`
**Delete Iteration**| `deliter INDEX`<br> e.g., `deliter 1`
**List customers**| `list`

> **Find customers**| `find [k/KEYWORD]... -all [t/TAG]... -any [t/TAG]...` <br> e.g. 

Reviewer: ```suggestion
**Find customers**| `find [k/KEYWORD]... -all [t/TAG]... -any [t/TAG]...` <br> e.g. `find -all t/friend t/colleague`
```
